### PR TITLE
Document `Navigation.get_simple_path()` often returning non-optimal results

### PR DESCRIPTION
**Pull request by Calinou**

### CODE REVIEW

**Comment by timcode77** [comment]
Could I suggest that the disclaimer about non-optimal paths is also added to the Navigation2D's node description also? The main use case for this Node is to get shortest paths. If the user can't do that with the node, we should let them know about it in the node description, not just the get_shortest_path() description.

**Comment by Calinou** [comment]
> Could I suggest that the disclaimer about non-optimal paths is also added to the Navigation2D's node description? The main use case for this Node is to get shortest paths. If the user can't do that with the node, we should let them know about it in the node description, not just the get_simple_path() description.

Done.

File: doc/classes/Navigation.xml [hunk 1]
```diff
@@ -61,6 +62,7 @@
 			</argument>
 			<description>
 				Returns the path between two given points. Points are in local coordinate space. If [code]optimize[/code] is [code]true[/code] (the default), the agent properties associated with each [NavigationMesh] (radius, height, etc.) are considered in the path calculation, otherwise they are ignored.
+				[b]Note:[/b] This method has known issues and will often return nonoptimal paths.
```
Comment (aaronfranke) [inline doc/classes/Navigation.xml:65]:
```suggestion
				[b]Note:[/b] This method has known issues and will often return non-optimal paths.
```

File: doc/classes/Navigation2D.xml [hunk 2]
```diff
@@ -39,6 +40,7 @@
 			</argument>
 			<description>
 				Returns the path between two given points. Points are in local coordinate space. If [code]optimize[/code] is [code]true[/code] (the default), the path is smoothed by merging path segments where possible.
+				[b]Note:[/b] This method has known issues and will often return nonoptimal paths.
```
Comment (aaronfranke) [inline doc/classes/Navigation2D.xml:43]:
```suggestion
				[b]Note:[/b] This method has known issues and will often return non-optimal paths.
```

File: doc/classes/Navigation.xml [hunk 3]
```diff
@@ -5,6 +5,7 @@
 	</brief_description>
 	<description>
 		Provides navigation and pathfinding within a collection of [NavigationMesh]es. By default, these will be automatically collected from child [NavigationMeshInstance] nodes, but they can also be added on the fly with [method navmesh_add]. In addition to basic pathfinding, this class also assists with aligning navigation agents with the meshes they are navigating on.
+		[b]Note:[/b] The current navigation system has many known issues and will not always return optimal paths as expected.
```
Comment (akien-mga) [inline doc/classes/Navigation.xml:8]: Maybe mention that the issues are fixed in 4.0?

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.3.

**Pull request merged**